### PR TITLE
The first version of training mode position reset

### PR DESCRIPTION
**Pull request by BarrettCid**
Includes the first version of training mode position reset. Fix for inputNormalizer in game.cxx from FIXMESTEVE.

### CODE REVIEW

File: src/game/game.h [hunk 1]
```diff
@@ -31,7 +31,9 @@ enum gameinputs : unsigned short {
 	Kick = 0x2000,
 	Slash = 0x8000,
 	HSlash = 0x200,
-	Dust = 0x100
+	Dust = 0x100,
+    Reset = 1,
```
Comment (Ninetime) [inline src/game/game.h:35]: Unless there is an explicit requirement, we should probably use the same hex formatting. ie (0x1 & 0x8)

Comment (BarrettCid) [reply src/game/game.h:35]: No explicit requirement, just forgot. I will update the values and test before adding a commit.

File: src/overlay/overlay.h [hunk 2]
```diff
@@ -24,5 +24,16 @@ static bool show_ggpo_host = false;
...
+
+struct TrainingModeHelper {
+	static bool leftCorner;
+	static bool center;
+	static bool rightCorner;
+	static bool positionSwapped;
+	static bool paused;
+	static unsigned int previousInput;
+	static int* playerInput;
+};
```
Comment (adanducci) [inline src/overlay/overlay.h:37]: If it consists entirely of static variables, this seems decidedly less like it should be a struct and more like it should be a namespace.

File: src/overlay/overlay.h [hunk 3]
```diff
@@ -24,5 +24,16 @@ static bool show_ggpo_host = false;
 static bool show_ggpo_join = false;
 static bool show_character_select = false;
 static bool show_character_data = false;
+static bool show_position_reset = false;
+
+struct TrainingModeHelper {
+	static bool leftCorner;
+	static bool center;
+	static bool rightCorner;
```
Comment (adanducci) [inline src/overlay/overlay.h:32]: The mutual exclusivity of these variables worries me. It seems like an opportunity for an enum.

File: src/overlay/overlay.h (hunk 2 above)
Comment (BarrettCid) [reply src/overlay/overlay.h:37]: Good point, would you prefer it be in it's own file or can it live in overlay.h and be defined in overlay.cpp? I personally prefer it's own file but would like to know what project you would want it to exist in if that's the case.

File: src/overlay/overlay.h (hunk 3 above)
Comment (BarrettCid) [reply src/overlay/overlay.h:32]: Thanks for the input. I have updated a version using an enum and namespace that I will submit once the question of where the namespace should live is answered.

**Pull request merged**